src/lib.rs

```diff
@@ -120,6 +126,42 @@ impl Task {
     }
 }
 
+/// A reference to a task.
+#[derive(Clone, Copy)]
+enum TaskRef<'a> {
+    /// A waker that wakes up a future.
+    Waker(&'a Waker),
+
+    /// An unparker that wakes up a thread.
+    #[cfg(feature = "std")]
+    Unparker(&'a parking::Unparker),
+}
+
+impl TaskRef<'_> {
+    /// Tells if this task will wake up the other task.
+    fn will_wake(self, other: Self) -> bool {
+        match (self, other) {
+            (Self::Waker(a), Self::Waker(b)) => a.will_wake(b),
+            #[cfg(feature = "std")]
+            (Self::Unparker(_), Self::Unparker(_)) => {
+                // TODO: Use unreleased will_unpark API.
+                false
+            }
+            #[cfg(feature = "std")]
```

Comment:
tying this to `"std"` is probably not a good idea. I'd prefer using `#[allow(unused)]` instead or making all cases explicit.